extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -260,6 +261,7 @@ public String toString()
 
   private volatile DateTime firstRunTime;
   private volatile KafkaConsumer consumer;
+  private volatile boolean lifecycleStarted = false;
```

Comment:
How about adding a new `InitNotice`? It can be processed by the existing `exec`, so we don't have to add a new `initializationExec`.

Reply:
I would need to start the notice handler thread to use InitNotice, which I was trying to avoid doing (the idea of the patch was to keep the supervisor in a state similar to what it would have been in if it were stuck in the current lifecycle start, while allowing the rest of the lifecycle to start).

Reply:
> the idea of the patch was to keep the supervisor in a state similar to what it would have been in if it were stuck in the current lifecycle start, while allowing the rest of the lifecycle to start

Got it. Makes sense. 

The reason I commented about `InitNotice` was, I'm not sure `initializationExec` is needed. Maybe we can use `scheduledExec` instead and cancel the scheduledFuture once `tryInit` succeeds? Or we can submit a runnable task to `exec` which calls `RetryUtils.retry()`.

Reply:
Sure, I'll change to submit a runnable to exec for the init retries